packages/serverless/src/integrations/aws_lambda.ts

```diff
@@ -0,0 +1,235 @@
+   */
+  private timeoutWarning: boolean = false;
+
+  /**
+   * configured time in miliseconds.
+   */
+  private configuredTimeInMilliseconds: number;
+
+  /**
+   * @inheritDoc
+   */
+  public constructor(
+    options: { awslambda?: any; context?: any; timeoutWarning?: any; configuredTimeInMilliseconds?: number } = {},
+  ) {
+    this.awsContext = options.context;
+    this.timeoutWarning = options.timeoutWarning;
+    this.configuredTimeInMilliseconds = options.configuredTimeInMilliseconds as number;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(_: (callback: EventProcessor) => void, getCurrentHub: () => Hub): void {
+    // lambda-bootstrap
+    const lambdaBootstrap = require.main;
+
+    if (!this.awsContext && !lambdaBootstrap) {
+      return;
+    }
+
+    const nodeJs: string = 'node';
+
+    /** capturing this context */
+    const setupOnceContext = this;
+
+    /** rapid runtime instance */
+    const rapidRuntime = lambdaBootstrap?.children[0].exports;
+
+    /** handler that is invoked in case of unhandled and handled exception */
+    const originalPostInvocationError = rapidRuntime.prototype.postInvocationError;
+
+    this.getCurrentHub = getCurrentHub;
+
+    const hub = this.getCurrentHub && this.getCurrentHub();
+
+    /** configured time */
+    this.configuredTimeInMilliseconds = this.awsContext.getRemainingTimeInMillis();
+
+    const timeoutWarningBuffer: number = 1500;
+    const millisToSeconds: number = 1000;
+
+    /** checking if configured Time In Milliseconds is greater than timeout Warning Buffer */
+    if (this.configuredTimeInMilliseconds > timeoutWarningBuffer) {
+      const configuredTimeInSec = Math.floor(this.configuredTimeInMilliseconds / millisToSeconds);
+      const configuredTimeInMilli = Math.floor(configuredTimeInSec * millisToSeconds);
+
+      /** check timeout flag */
+      if (this.timeoutWarning === true) {
+        this.awsContext.callbackWaitsForEmptyEventLoop = false;
+        timeOutError(configuredTimeInMilli);
+      }
+    }
+
+    /**
+     * This function is invoked when there is timeout error
+     * Here, we make sure the error has been captured by Sentry Dashboard
+     * and then re-raise the exception
+     * @param configuredTime  - configured time in seconds
+     * @hidden
+     */
+    function timeOutError(configuredTime: number): void {
+      setTimeout(() => {
+        /**
+         * setting parameters in scope which will be displayed as additional data in Sentry dashboard
+         */
+        setParameters();
+
+        const convertTimeToSeconds = Math.floor(configuredTime / millisToSeconds);
+
+        const error = new Error(
+          `WARNING : Function is expected to get timed out. Configured timeout duration = ${
+            convertTimeToSeconds + 1
+          } seconds.`,
+        );
+
+        /** capturing the exception and re-directing it to the Sentry Dashboard */
+        hub.captureException(error);
+        Sentry.flush(2000);
```

Comment:
It shouldn't be fixed value. Integration should allow for configuration here as well.

Reply:
made this change in the new pull request.